app/models/spotlight/resource.rb

```diff
@@ -61,9 +61,7 @@ def becomes_provider
       klass = Spotlight::Resource.class_for_resource(self)
 
       if klass
-        z = self.becomes klass
-        z.type = z.class.to_s
-        z
+        self.becomes! klass
```

Comment:
@jkeck thanks for finding `#becomes!`, it seems like it's exactly what we needed.